sender_test.go

```diff
@@ -2,32 +2,769 @@ package amqp
+		switch tt := req.(type) {
+		case *mocks.AMQPProto:
+			return []byte{'A', 'M', 'Q', 'P', 0, 1, 0, 0}, nil
+		case *frames.PerformOpen:
+			return mocks.PerformOpen("container")
+		case *frames.PerformBegin:
+			return mocks.PerformBegin(0)
+		case *frames.PerformEnd:
+			return mocks.PerformEnd(0, nil)
+		case *frames.PerformAttach:
+			mode := encoding.ModeUnsettled
+			return mocks.EncodeFrame(mocks.FrameAMQP, 0, &frames.PerformAttach{
+				Name:   tt.Name,
+				Handle: 0,
+				Role:   encoding.RoleReceiver,
+				Target: &frames.Target{
+					Address:      "test",
+					Durable:      encoding.DurabilityNone,
+					ExpiryPolicy: encoding.ExpirySessionEnd,
+				},
+				SenderSettleMode: &mode,
+				MaxMessageSize:   16, // really small messages only
+			})
+		case *frames.PerformTransfer:
+			return mocks.PerformDisposition(encoding.RoleReceiver, 0, *tt.DeliveryID, &encoding.StateAccepted{})
+		case *frames.PerformDetach:
+			return mocks.PerformDetach(0, 0, nil)
+		default:
+			return nil, fmt.Errorf("unhandled frame %T", req)
+		}
+	}
+	netConn := mocks.NewNetConn(responder)
+
+	client, err := New(netConn)
+	require.NoError(t, err)
+
+	session, err := client.NewSession()
+	require.NoError(t, err)
+	time.Sleep(100 * time.Millisecond)
+	snd, err := session.NewSender()
+	require.NoError(t, err)
+
+	sendInitialFlowFrame(t, netConn, 0, 100)
+	time.Sleep(100 * time.Millisecond)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	require.Error(t, snd.Send(ctx, NewMessage([]byte("test message that's too big"))))
+	cancel()
+
+	time.Sleep(100 * time.Millisecond)
+	require.NoError(t, client.Close())
+}
+
+func TestSenderSendTagTooBig(t *testing.T) {
+	responder := func(req frames.FrameBody) ([]byte, error) {
+		switch tt := req.(type) {
+		case *mocks.AMQPProto:
+			return []byte{'A', 'M', 'Q', 'P', 0, 1, 0, 0}, nil
+		case *frames.PerformOpen:
+			return mocks.PerformOpen("container")
+		case *frames.PerformBegin:
+			return mocks.PerformBegin(0)
+		case *frames.PerformEnd:
+			return mocks.PerformEnd(0, nil)
+		case *frames.PerformAttach:
+			return mocks.SenderAttach(0, tt.Name, 0, encoding.ModeUnsettled)
+		case *frames.PerformTransfer:
+			return mocks.PerformDisposition(encoding.RoleReceiver, 0, *tt.DeliveryID, &encoding.StateAccepted{})
+		case *frames.PerformDetach:
+			return mocks.PerformDetach(0, 0, nil)
+		default:
+			return nil, fmt.Errorf("unhandled frame %T", req)
+		}
+	}
+	netConn := mocks.NewNetConn(responder)
+
+	client, err := New(netConn)
+	require.NoError(t, err)
+
+	session, err := client.NewSession()
+	require.NoError(t, err)
+	time.Sleep(100 * time.Millisecond)
+	snd, err := session.NewSender()
+	require.NoError(t, err)
+
+	sendInitialFlowFrame(t, netConn, 0, 100)
+	time.Sleep(100 * time.Millisecond)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	msg := NewMessage([]byte("test"))
+	msg.DeliveryTag = make([]byte, 33)
```

Comment:
This is the most important line in the test. Maybe a _tiny_ bit of commentary or something would help point it out.